contracts/Airdrop.sol

```diff
@@ -0,0 +1,238 @@
+pragma solidity 0.8.10;
+
+import "./interfaces/IAirdrop.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+
+/// @title Airdrop
+/// @author Applicature
+/// @notice This smart contract is used for sending coins or tokens to wallet addresses
+/// @dev This smart contract is used for sending coins or tokens to wallet addresses
+contract Airdrop is IAirdrop, Ownable, EIP712 {
+    using SafeERC20 for IERC20;
+
+    /// @notice the address of reward token
+    /// @dev the address of reward token
+    IERC20 public rewardToken;
+
+    /// @notice the accrued reward tokens per users
+    /// @dev the accrued reward tokens per users
+    mapping(address => uint256) public userTokens;
+
+    /// @notice the accrued ether per users
+    /// @dev the accrued ether per users
+    mapping(address => uint256) public userEthers;
+
+    /// @notice Store nonces of recipients
+    /// @dev Store nonces of recipients
+    mapping(address => mapping(uint256 => bool)) internal _nonces;
+
+    /// @notice Store hash to sign drop transaction
+    /// @dev Store computed 256 bit keccak hash
+    bytes32 private constant _CONTAINER_TYPEHASE =
+        keccak256(
+            "Container(address sender,address[] recipients,uint256[] amounts,uint256 deadline,uint256 nonce)"
+        );
+
+    /// #if_succeeds {:msg "the reward token must be initialized correctly"} rewardToken == IERC20(_token);
+    /// @notice Initialize contract
+    /// @dev Initialize contract, sets reward token address
+    /// @param _token the reward token address
+    constructor(address _token) EIP712("Airdrop", "v1") {
+        require(_token != address(0), "ERROR_INVALID_ADDRESS");
+        rewardToken = IERC20(_token);
+    }
+
+    /// #if_succeeds {:msg "the balance must increase by amount"} rewardToken.balanceOf(address(this)) ==
+    /// old(rewardToken.balanceOf(address(this)) + _amount);
+    /// @notice Transfers tokens from owner to smart contract
+    /// @dev Transfers tokens from owner to smart contract
+    /// @param _amount the amount of tokens
+    function depositTokens(uint256 _amount) external override onlyOwner {
+        require(_amount > 0, "ERROR_WRONG_AMOUNT");
+        rewardToken.safeTransferFrom(msg.sender, address(this), _amount);
+        emit DepositTokens(msg.sender, _amount);
+    }
+
+    /// #if_succeeds {:msg "the balance ether must increase by amount"} address(this).balance == old(address(this).balance + msg.value);
+    /// @notice Transfers ethers from owner to smart contract
+    /// @dev Transfers ethers from owner to smart contract
+    function depositEther() external payable override onlyOwner {
+        emit DepositEther(msg.sender, msg.value);
+    }
+
+    /// #if_succeeds {:msg "nonce must be non-working"} _nonces[msg.sender][_nonce] == true;
```

Comment:
if I'm not mistaken, the next test for you should be the description "nonce should be broken": you need to check old (_nonces) for this description but you check new values, which will always be true after passing the method